play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
```

Comment:
I would call it `MacroOptionsDefault` instead of `DefaultMacroOptions`. `MacroOptionsDefault` suggests it's a new type of thing, as opposed to a default implementation of `MacroOptions`.

Instead of `NotDefaultMacroOptions` I think it's ok to call it `LowPriorityMacroOptionsDefaultImplicits`. The name is more verbose but the "LowPriority*" makes it more readable since it follows the standard naming convention for this type of trait.

@jroper It would also be useful to have an explanation like you provided in the comments here about how this works to provide an appropriate default value for `JsonConfiguration`.